sklearn/datasets/_arff_parser.py

```diff
@@ -371,11 +372,28 @@ def _pandas_arff_parser(
         names=[name for name in openml_columns_info],
         dtype=dtypes,
     )
-
     columns_to_select = feature_names_to_select + target_names_to_select
     columns_to_keep = [col for col in frame.columns if col in columns_to_select]
     frame = frame[columns_to_keep]
 
+    # strip quotes to be consistent with LIAC ARFF
+    re_start_end_quotes = "^[\"'](.*[\"'])$"
+
+    def strip_quotes(s):
+        return s.group(0)[1:-1]
+
+    for col in make_column_selector(dtype_exclude="number")(frame):
+        if pd.api.types.is_categorical_dtype(frame[col].dtype):
+            # modify the categories instead of each dataframe row
+            frame[col].cat.categories = frame[col].cat.categories.str.replace(
+                re_start_end_quotes, strip_quotes, regex=True
+            )
+        else:
+            # we deal with an object column
+            frame[col] = frame[col].replace(
+                re_start_end_quotes, strip_quotes, regex=True
+            )
```

Comment:
From my reading of the [arff format](https://www.cs.waikato.ac.nz/ml/weka/arff.html), I do not think there are object dtypes besides dates, which should not have the quote issue. The "Nominal" type should all be converted into categoricals.

Are there datasets where there is a object dtype with the pandas parser?

Reply:
Titanic for the "name" column

Reply:
Ah, I see there is a "string" attribute, which would end up to be an object dtype.

We likely need to handle this case. To avoid the copy, we can check if there are quotes and only copy if it is required.

Reply:
Do you mean to make the following:

```python
if frame[col].str.match(re_start_end_quotes).any():
    # make the replace in this case
```